<a href="https://colab.research.google.com/github/ericakitano/Project-2/blob/main/Project_2_Part_1_(Core)_Erica_Kitano.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment:

Create a notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:



In [1]:
#Import Libraries
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# First choice: dataset 1

In [3]:
#Load Data
filename1 = '/content/drive/MyDrive/Colab Notebooks/Raw Data Files/Project/Project2/healthcare-dataset-stroke-data.csv'
df1 = pd.read_csv (filename1)
df1.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [4]:
df1.shape

(5110, 12)

In [5]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [9]:
df1['stroke'].value_counts()

0    4861
1     249
Name: stroke, dtype: int64

In [11]:
df1.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

## First choice: dataset 1
1. Source of data: [Stroke Prediction Dataset](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)

2. Brief description of data: 

   > This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.
   > 
   > Source: https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

3. What is the target?

   > column: 'stroke'

4. What does one row represent? (A person?  A business?  An event? A product?)

   > Patient (person)

5. Is this a classification or regression problem?

   > Classification

6. How many features does the data have?

   > Currently there are 11 other columns besides the target column. 
   >
   > However, after dropping unnnecessary column (e.g. 'id'), the number of feature columns will reduce to 10. 
   >
   > There are 10 features.

7. How many rows are in the dataset?

   > 5110

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
   > 'bmi' column is missing 201 values.



***


# Second choice: dataset 2

In [6]:
#Load Data
filename2 = '/content/drive/MyDrive/Colab Notebooks/Raw Data Files/Project/Project2/Metabolic_Syndrome.csv'
df2 = pd.read_csv (filename2)
df2.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [7]:
df2.shape

(2401, 15)

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


In [13]:
df2['MetabolicSyndrome'].value_counts()

No MetSyn    1579
MetSyn        822
Name: MetabolicSyndrome, dtype: int64

In [14]:
df2.isna().sum()

seqn                   0
Age                    0
Sex                    0
Marital              208
Income               117
Race                   0
WaistCirc             85
BMI                   26
Albuminuria            0
UrAlbCr                0
UricAcid               0
BloodGlucose           0
HDL                    0
Triglycerides          0
MetabolicSyndrome      0
dtype: int64

## Second choice: dataset 2
1. Source of data: [Metabolic Syndrome Prediction](https://data.world/informatics-edu/metabolic-syndrome-prediction)

2. Brief description of data

   > To predict metabolic syndrome, yes or not based on common risk factors
   >
   > Source: https://data.world/informatics-edu/metabolic-syndrome-prediction
   
3. What is the target?

   > MetablicSyndrome

4. What does one row represent? (A person?  A business?  An event? A product?)

   > A person

5. Is this a classification or regression problem?

   > Classification

6. How many features does the data have?

   > 14

7. How many rows are in the dataset?

   > 2401

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

  > There are missing values in 4 columns: Marital, Income, WaistCirc, BMI.